core/services/job/models_test.go

```diff
@@ -71,3 +81,179 @@ func TestOCR2OracleSpec_RelayIdentifier(t *testing.T) {
 		})
 	}
 }
+
+var (
+	//go:embed testdata/compact.toml
+	compact string
+	//go:embed testdata/pretty.toml
+	pretty string
+)
+
+func TestOCR2OracleSpec(t *testing.T) {
+	val := OCR2OracleSpec{
+		Relay:                             relay.EVM,
+		PluginType:                        types.Median,
+		ContractID:                        "foo",
+		OCRKeyBundleID:                    null.StringFrom("bar"),
+		TransmitterID:                     null.StringFrom("baz"),
+		ContractConfigConfirmations:       1,
+		ContractConfigTrackerPollInterval: *models.NewInterval(time.Second),
+		RelayConfig: map[string]interface{}{
+			"chainID":   1337,
+			"fromBlock": 42,
+			"chainReader": evmtypes.ChainReaderConfig{
+				Contracts: map[string]evmtypes.ChainContractReader{
+					"median": {
+						ContractABI: `[
+  {
+    "anonymous": false,
+    "inputs": [
+      {
+        "indexed": true,
+        "internalType": "address",
+        "name": "requester",
+        "type": "address"
+      },
+      {
+        "indexed": false,
+        "internalType": "bytes32",
+        "name": "configDigest",
+        "type": "bytes32"
+      },
+      {
+        "indexed": false,
+        "internalType": "uint32",
+        "name": "epoch",
+        "type": "uint32"
+      },
+      {
+        "indexed": false,
+        "internalType": "uint8",
+        "name": "round",
+        "type": "uint8"
+      }
+    ],
+    "name": "RoundRequested",
+    "type": "event"
+  },
+  {
+    "inputs": [],
+    "name": "latestTransmissionDetails",
+    "outputs": [
+      {
+        "internalType": "bytes32",
+        "name": "configDigest",
+        "type": "bytes32"
+      },
+      {
+        "internalType": "uint32",
+        "name": "epoch",
+        "type": "uint32"
+      },
+      {
+        "internalType": "uint8",
+        "name": "round",
+        "type": "uint8"
+      },
+      {
+        "internalType": "int192",
+        "name": "latestAnswer_",
+        "type": "int192"
+      },
+      {
+        "internalType": "uint64",
+        "name": "latestTimestamp_",
+        "type": "uint64"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  }
+]
+`,
```

Comment:
I don't think this is being read right, I wasn't thinking about the config being serialized (only deserialized), so the type param should be missing when it gets serialized and read in again.  I'll open a PR (with a test) to fix, but I think this config isn't being read either or it would return an error.

Reply:
Yeah it looks like `ModifiersConfig` implements only `UnmarshalJSON` but needs a `MarshalJSON` equivalent to write the types.

Reply:
I pushed a hacky wrapper that seems to work, if you are looking for inspiration.